relbench/data/task_base.py

```diff
@@ -40,10 +37,7 @@ def __init__(
                 f"diff: {time_diff})."
             )
 
-        self.target_col = target_col
         self.metrics = metrics
-        self.entity_table = entity_table
-        self.entity_col = entity_col
 
         self._full_test_table = None
         self._cached_table_dict = {}
```

Comment:
Maybe in this `BaseTask`, you can prepare `set_cached_table_dict` or something and call it in your `RelBenchNodeTask` and `RelBenchLinkTask` to set the cache (if any)